Pidaq/Lib/uploadToDrive/uploadToDrive.py

```diff
@@ -0,0 +1,125 @@
+from __future__ import print_function
+import os
+import logging
+from datetime import datetime
+import json
+
+from googleapiclient.discovery import build
+from httplib2 import Http
+from oauth2client import file, client, tools
+
+
+SCOPES = 'https://www.googleapis.com/auth/drive'
+store = file.Storage('credentials.json') 
+creds = store.get()
+
+if not creds or creds.invalid:
+    flow = client.flow_from_clientsecrets('client_secret.json', SCOPES)
+    creds = tools.run_flow(flow, store)
+    
+DRIVE = build('drive', 'v3', http=creds.authorize(Http()))
+
+# Searches for the Json file containing folder names and IDS
+# These IDs are to be stored manually.
+try:
+    with open('dictionary.json') as json_file:    
+        parentDictionary = json.load(json_file)
+except:
+    logging.error('Parents dictionary .json file not found')
+    exit()
+# change this for the ID of the default folder 
+DefaultID = '1VDOOjuOeyNRJgdKmmRTJC1HVjq91F4Q8'
+
+def uploadCsv(filePath, parentName):
+    """ Uploads a csv file from a local directory to a folder on Google Drive.
+
+    Parameters:
+    "filePath" is the path of the csv file itself and
+                should be passed without quotations.
+    "parentName" is the name of the testing folder in the drive
+            and only a specific set of values are allowed.
+    The parent folder is supposedly going to be fixed for a
+    long period of time.
+    """
+    # searches for the parent name in the dictionary
+    for key in parentDictionary.keys():
+        if key.lower() == parentName.lower():
+            parentKey = key
+    try:
+        parentFolderID = parentDictionary[parentKey]
+    except KeyError:
+        # uploads to a default folder if the folder name does
+        # not exist in the dictionary
+        parentFolderID = DefaultID
+        logging.error('Folder name not found \nuploading to "Default"')

```

Comment:
2 things:
1.) In the first for loop - you would generally break after you find what you are looking for. Also - in the case where the key is not in the dictionary - `parentKey` never gets defined and you will get an error you arent looking for when you try `parentDictionary[parentKey]`
2.) Its a little weird how you iterate through all the keys first, looking for a matching key. And then afterwards - you check again if it exists (and in this case if the key doesnt exist - `parentKey` doesnt get initialized and an error will occur.

Couple of solutions here - not sure what we want to do for sure.

* Instead of iterating through all the keys and fixing the case there - when you load in the dict you could .lower() all the keys like:
```
json_dict = json.load()
parentDictionary = {}
for key, value in json_dict.items():
    parentDictionary[key.lower()] = value
```
 and then just try `parentDictionary[parentKey.lower()]`. This way we the json can contain be either uppercase or lowercase folder names as keys, and we will still find it when we do a lookup.

This is probably not the best solution as we are copying a whole dict for not much reason.

* We could simply say all folder names have to be all lower case or all upper case in the json, and just do the corresponding `.lower()` or `.upper()` to the key that we pass to the function.

* Finally, it really isn't that deep. We could also just not worry about case - if the key isn't there exactly as you pass it, an error is thrown and you fix your call to the function - easy as that. The important thing is you catch the case where there is no corresponding folder.

Reply:
yeah, it should have been NameError as `parentKey` would not be defined in this case. I think the way Colton is suggesting would be better. All upper or all lower might be something we want to specify in the README that's about to get uploaded, but we wouldn't have to if this works